source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -145,10 +145,17 @@ OptRef<const LocalRateLimiterImpl::LocalDescriptorImpl> LocalRateLimiterImpl::de
 
 bool LocalRateLimiterImpl::requestAllowed(
     absl::Span<const RateLimit::LocalDescriptor> request_descriptors) const {
-  auto descriptor = descriptorHelper(request_descriptors);
-
-  return descriptor.has_value() ? requestAllowedHelper(*descriptor.value().get().token_state_)
-                                : requestAllowedHelper(tokens_);
+  if (!descriptors_.empty() && !request_descriptors.empty()) {
+    bool limit = true;
+    for (const auto& request_descriptor : request_descriptors) {
+      auto it = descriptors_.find(request_descriptor);
+      if (it != descriptors_.end()) {
+        limit &= requestAllowedHelper(*it->token_state_);
+      }
+    }
+    return limit;
+  }
+  return requestAllowedHelper(tokens_);
```

Comment:
If we try to consume token from all the descriptors, should we consume token from the global `tokens_`?

Reply:
thanks, i think global `tokens_` should also be consumed.